### ISSUE TITLE
Support backup mirrors for software repos

### ISSUE THREAD

**Issue by nchaimov** [ParaToolsInc/taucmdr#321]
The FSF server is down currently, which is preventing TAU Commander from being able to download binutils, which in turn prevents TAU Commander from installing. TAU Commander only has one URL to try for binutils (or for other dependencies). We should support multiple mirrors for repos, so that we can fall back on a backup should the main repo be down.

**Comment by zbeekman** [comment]
Are there existing mirrors, or shall we mirror to our own infra/github? I have a github action that can do this pretty easily.

**Comment by nchaimov** [comment]
There are a bunch of mirrors for FSF (where we get binutils from): https://www.gnu.org/prep/ftp.en.html

I doesn't look like there is a mirror for PAPI. We should possibly mirror ourselves.

TAU, PDT, Score-P, OMPT, libunwind, libotf2 off of uoregon infrastructure. We should probably have a mirror for this. On GitHub itself might be good. If we mirror it on, say, fs.paratools.com, then a UO network outage would bring down the original and the mirror.

**Comment by nchaimov** [comment]
This is implemented in PR #324, although merging of that is waiting on fixing an issue with the Travis testing. Testing succeeds locally, but building the `tau_trace2json` command line tool is failing on Travis for some reason.

**Comment by zbeekman** [comment]
Yes, anything open source we should mirror on GH, IMO, for speed & redundancy.

**Comment by khsa1** [comment]
Capability added in 1024368b8f536d5d25973279fedd800abdaeb953.

We still need to add the backup mirrors.

**Comment by zbeekman** [comment]
I'm going to gauge interest with Kevin, Sameer and the rest of the TAU team for setting up a mirror and maybe even tagged releases for all the required upstream packages (TAU, PAPI, PDT, binutils, etc.). It would certainly be nice to have greater availability for TAU and dependencies. There is a TAU mirror at https://github.com/UO-OACISS/tau2 which might be a natural place for hosting binaries/releases.

**Comment by zbeekman** [comment]
There is now a cron job that puts the mirrored packages at http://fs.paratools.com/tau-mirror/ I'll update the sources to include the mirrors and hit them if/when downloads fail.
